components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -203,6 +274,12 @@ private void populateFeatureAttributeMapping() throws Exception {
         }
     }
 
+    private void logError(int modelId, Exception e) {
+        log.error("Error while retrieving ML-model : " + modelStorageLocations[modelId], e);
+        throw new ExecutionPlanCreationException(
```

Comment:
Yes. `ExecutionPlanCreationException` is a runtime exception.
How does it need to be handled we need to throw this runtime exception for all 4 exceptions we catch?